### PR TITLE
Add capability to pass blob versioned hashes to engine_newPayloadV3

### LINKED ISSUE #7203: Use engine_newPayloadV3 to pass versioned_hashes to EL for validation
https://github.com/ethereum/consensus-specs/pull/3359

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
This PR mostly changes the interfaces and introduces `NewPayloadRequest` as in the spec.

- Add `blobVersionedHashes` as second parameter to `engine_newPayloadV3`

In next PR, will pass the real versioned hashes in `ForkChoicePayloadExecutor`

## Fixed Issue(s)
will help with #7203

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by tbenr** [comment]
the keyword `resolve` in the description will close the issue..

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/JsonRpcRequestParams.java [hunk 1]
```diff
@@ -51,6 +51,11 @@ public <T> Optional<T> getOptionalParameter(final int index, final Class<T> para
     return Optional.of((T) param);
   }
 
+  @SuppressWarnings({"unchecked", "unused"})
```
Comment (tbenr) [inline ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/JsonRpcRequestParams.java:54]: `unused`?

**Comment by StefanBratanov** [comment]
> the keyword `resolve` in the description will close the issue..

Oh yes, thanks!

File: ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java [hunk 2]
```diff
@@ -197,7 +203,49 @@ public void exchangeCapabilities_shouldBuildRequestAndResponseSuccessfully() thr
...
+    assertThat(futureResponse)
+        .succeedsWithin(1, TimeUnit.SECONDS)
+        .matches(
+            response ->
+                response.getPayload().asInternalExecutionPayload().equals(payloadStatusResponse));
+
+    final Map<String, Object> requestData = takeRequest();
+    verifyJsonRpcMethodCall(requestData, "engine_newPayloadV3");
+
+    assertThat(((List<Object>) requestData.get("params")).get(1))
```
Comment (tbenr) [inline ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java:249]: annoying but I think we could check `((List<Object>) requestData.get("params")).get(0)` as well, derived from executionPayloadV3

**Review by tbenr — APPROVED** [review]
LGTM with an nit and an improvement

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/JsonRpcRequestParams.java (hunk 1 above)
Comment (StefanBratanov) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/JsonRpcRequestParams.java:54]: Essentially the Class parameter in the method is unused but helps the compiler understand what type is returned, so can pipe orElse() afterwards.

File: ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java (hunk 2 above)
Comment (StefanBratanov) [reply ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java:249]: Yeah, makes sense. Will tackle tomorrow.

Comment (StefanBratanov) [reply ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java:249]: Actually just done it.

**Pull request merged**